onadata/libs/utils/dict_tools.py

```diff
@@ -94,3 +96,42 @@ def csv_dict_to_nested_dict(a):
     merged_dict = merge_list_of_dicts(results)
 
     return remove_indices_from_dict(merged_dict)
+
+
+def dict_lists2strings(d):
+    """Convert lists in a dict to joined strings.
+
+    :param d: The dict to convert.
+    :returns: The converted dict."""
+    for k, v in d.items():
+        if isinstance(v, list) and all([isinstance(e, basestring) for e in v]):
+            d[k] = ' '.join(v)
+        elif isinstance(v, dict):
+            d[k] = dict_lists2strings(v)
+
+    return d
+
+
+def dict_paths2dict(d):
+    result = {}
```

Comment:
This function is missing a docstring. So line 124 is creating a nested dict?  think an example input and output in the docstring would help clarify how it's working